agent/api/task/task.go

```diff
@@ -940,32 +949,36 @@ func (task *Task) getAllASMSecretRequirements() map[string]apicontainer.Secret {
 }
 
 // GetFirelensContainer returns the firelens container in the task, if there is one.
-func (task *Task) GetFirelensContainer() *apicontainer.Container {
+func (task *Task) GetFirelensContainers() []*apicontainer.Container {
+	var firelensContainers []*apicontainer.Container
 	for _, container := range task.Containers {
 		if container.GetFirelensConfig() != nil { // This is a firelens container.
-			return container
+			firelensContainers = append(firelensContainers, container)
 		}
 	}
-	return nil
+	return firelensContainers
 }
 
 // initializeFirelensResource initializes the firelens task resource and adds it as a dependency of the
 // firelens container.
 func (task *Task) initializeFirelensResource(config *config.Config, resourceFields *taskresource.ResourceFields,
-	firelensContainer *apicontainer.Container, credentialsManager credentials.Manager) error {
-	if firelensContainer.GetFirelensConfig() == nil {
-		return errors.New("firelens container config doesn't exist")
+	firelensContainers []*apicontainer.Container, credentialsManager credentials.Manager) error {
+	if len(firelensContainers) == 0 {
+		return errors.New("unable to initialize firelens resource because there's no firelens container")
 	}
-
 	containerToLogOptions := make(map[string]map[string]string)
-	// Collect plain text log options.
-	if err := task.collectFirelensLogOptions(containerToLogOptions); err != nil {
-		return errors.Wrap(err, "unable to initialize firelens resource")
-	}
+	firelensVersion := firelensContainers[0].FirelensConfig.Version
```

Comment:
can there be v1 and v2 firelens containers in the same task?

Reply:
Nope, a task can only have either firelens v1 or firelens v2 container(s).